Jinja2>=3.0.0 makes the docs build fail

### Environment Details

* SDV version: 0.10.0

### Error Description

The docs build for v0.10.0 failed because of [this error](https://github.com/pydata/pydata-sphinx-theme/issues/394):

```
generating indices...  genindexfailed

Theme error:
An error happened in rendering the page genindex.
Reason: UndefinedError("'meta' is undefined")
make[1]: *** [Makefile:20: html] Error 2
```

The error seems to happen with `Jinja2>=3.0.0`, which was released on May 11th, 2021.

We should cap the Jinja2 version to `<3` to fix the problem, and this should be done directly on the `stable` branch as a hotfix to fix the current docs build.

--- a/setup.py
+++ b/setup.py
@@ -52,6 +52,9 @@ development_requires = [
     'Sphinx>=3,<3.3',
     'pydata-sphinx-theme<0.5',
 
+    # Jinja2>=3 makes the sphinx theme fail
+    'Jinja2>=2,<3',
+
     # style check
     'flake8>=3.7.7,<4',
     'flake8-absolute-import>=1.0,<2',

